### PR TITLE
Improve auto-fix for `vue/define-macros-order` rule

### PR DESCRIPTION
**Pull request by edikdeisling**
In PR https://github.com/vuejs/eslint-plugin-vue/pull/1861 was decided to create separate PR for these improvements

There are two fixes here:
1. Fix extra newlines appear
2. Fix corner case `a();b() -> b()a();`

### CODE REVIEW

File: lib/rules/define-macros-order.js [hunk 1]
```diff
@@ -232,13 +232,12 @@ function create(context) {
     const targetComment = sourceCode.getTokenAfter(beforeTargetToken, {
       includeComments: true
     })
-    const textSpace = getTextBetweenTokens(beforeTargetToken, targetComment)
```
Comment (edikdeisling) [inline lib/rules/define-macros-order.js:235]: This thing fixes these newlines https://github.com/vuejs/eslint-plugin-vue/pull/1861/files#diff-6ebdd3c21be70e29ce94a2f399884fd220d4fe9bba358ade66b0bceb325d3102L158

Before we took space **before** the target token and added it after the moving token.
Now we take space **after** the target token and added it after the moving token.

File: lib/rules/define-macros-order.js [hunk 2]
```diff
@@ -247,17 +246,28 @@ function create(context) {
...
+  function getInsertText(textNode, target) {
+    const afterTargetComment = sourceCode.getTokenAfter(target, {
+      includeComments: true
+    })
+    const afterText = sourceCode.text.slice(
+      target.range[1],
+      afterTargetComment.range[0]
+    )
+    // handle case when a();b() -> b()a();
+    const invalidResult = !textNode.endsWith(';') && !afterText.includes('\n')
```
Comment (edikdeisling) [inline lib/rules/define-macros-order.js:262]: This is corner case `a();b() -> b()a();`.
`afterText` contains symbols between target token(`a();`) and next token(`b()`). If there are no new lines and `b()` doesn't ends with `;` - we add `;` manually

**Comment by edikdeisling** [comment]
I ran it in my project and the result was great. The only annoying thing left is this:
image
There is an option to introduce "newLineAfterDefineMacros: true" option maybe... For now, I've done nothing

File: lib/rules/define-macros-order.js (hunk 2 above)
Comment (FloEdelmann) [reply lib/rules/define-macros-order.js:262]: Can you please add a test case for this edge case?

**Review by FloEdelmann — CHANGES_REQUESTED** [review]

File: lib/rules/define-macros-order.js (hunk 2 above)
Comment (edikdeisling) [reply lib/rules/define-macros-order.js:262]: Sure. Done

**Review by FloEdelmann — APPROVED** [review]
Thanks!

**Review by ota-meshi — APPROVED** [review]
LGTM!

**Comment by ota-meshi** [comment]
> There is an option to introduce "newLineAfterDefineMacros: true" option maybe... For now, I've done nothing

I think we shouldn't add options for auto-fix. Because the aims of the rule is to check.
